x-pack/plugins/observability_solution/entity_manager/server/lib/entities/helpers/generate_component_id.ts

```diff
@@ -38,6 +39,10 @@ export function generateHistoryIndexName(definition: EntityDefinition) {
   });
 }
 
+export function generateHistoryIndexTemplateId(definition: EntityDefinition) {
+  return `${ENTITY_BASE_PREFIX}_${ENTITY_SCHEMA_VERSION_V1}_${ENTITY_HISTORY}_${definition.id}_index_template` as const;
```

Comment:
Totally out of scope for this PR:
This might be me being pedantic, but our other ES assets use dash-case, if it's not too much effort, could we change the index and component templates to follow the same pattern? 
Similar, we don't apply any prefixes to the names of the component templates currently.